pkg/oauth/tokenrequest/request_token.go

```diff
@@ -284,6 +291,71 @@ func (o *RequestTokenOptions) RequestToken() (string, error) {
 	}
 }
 
+func (o *RequestTokenOptions) RequestTokenWithAuthorization(authorizeUrlHandler func(string) error, callbackPort int) (string, error) {
+	if o.OsinConfig == nil {
+		if err := o.SetDefaultOsinConfig(); err != nil {
+			return "", err
+		}
+	}
+
+	callbackServer, err := newCallbackServer(callbackPort)
+
+	listenAddr, ok := callbackServer.tcpListener.Addr().(*net.TCPAddr)
+	if !ok {
+		return "", fmt.Errorf("listener is not of TCP type: %v", callbackServer.tcpListener.Addr())
+	}
+
+	o.OsinConfig.ClientId = openShiftCLIBrowserClientID
+	o.OsinConfig.RedirectUrl = fmt.Sprintf("http://%s/callback", listenAddr)
+
+	client, err := newOsinClient(o)
+	if err != nil {
+		return "", err
+	}
+
+	authorizeRequest := client.NewAuthorizeRequest(osincli.CODE)
+	if err := authorizeUrlHandler(authorizeRequest.GetAuthorizeUrl().String()); err != nil {
+		return "", err
+	}
+
+	callbackServer.SetCallbackHandler(func(callback *http.Request) (string, error) {
```

Comment:
this should be a part of the constructor

Reply:
The reason I didn't add it in the constructor is because `requestAccessToken` requires the `client` and `authorizeRequest`, and in the way I've defined it now it can access both from the local scope.

I could move it to the constructor, but I would need to provide a setter method for the two parameters; I tend to prefer the closure setter as it provides more clarity as to what it does, what do you think?